profiler/src/main/java/com/splunk/opentelemetry/profiler/BasicJfrRecordingFile.java

```diff
@@ -38,21 +42,33 @@ public BasicJfrRecordingFile(JFR jfr) {
   public Stream<RecordedEvent> open(Path path) {
     RecordingFile file = jfr.openRecordingFile(path);
     return StreamSupport.stream(
-        new Spliterators.AbstractSpliterator<RecordedEvent>(Long.MAX_VALUE, Spliterator.ORDERED) {
-          public boolean tryAdvance(Consumer<? super RecordedEvent> action) {
-            if (file.hasMoreEvents()) {
-              action.accept(jfr.readEvent(file, path));
-              return true;
-            }
-            return false;
-          }
-
-          public void forEachRemaining(Consumer<? super RecordedEvent> action) {
-            while (file.hasMoreEvents()) {
-              action.accept(jfr.readEvent(file, path));
-            }
-          }
-        },
-        false);
+            new Spliterators.AbstractSpliterator<RecordedEvent>(
+                Long.MAX_VALUE, Spliterator.ORDERED) {
+              public boolean tryAdvance(Consumer<? super RecordedEvent> action) {
+                if (file.hasMoreEvents()) {
+                  action.accept(jfr.readEvent(file, path));
+                  return true;
+                }
+                closeSafely(file);
+                return false;
+              }
+
+              public void forEachRemaining(Consumer<? super RecordedEvent> action) {
+                while (file.hasMoreEvents()) {
+                  action.accept(jfr.readEvent(file, path));
+                }
+                closeSafely(file);
```

Comment:
Nit: you probably don't need to call `closeSafely()` here and in `tryAdvance()` now that you've added it as the close handler.

Reply:
The intention was to be helpful in the event that the user doesn't actually call close.  Best effort close() is called for you if you consume the entire stream.

Do you think it's confusing to have close in here and ask the caller to to also call close?  Want me to remove it?

Reply:
Hmm, I guess it's probably fine to leave it if doesn't throw on second call